core/src/main/java/com/google/zxing/qrcode/decoder/DecodedBitStreamParser.java

```diff
@@ -111,16 +111,16 @@ static DecoderResult decode(byte[] bytes,
             int count = bits.readBits(mode.getCharacterCountBits(version));
             switch (mode) {
               case NUMERIC:
-                decodeNumericSegment(bits, result, count);
+                decodeNumericSegment(bits, result, count, byteSegments);
                 break;
               case ALPHANUMERIC:
-                decodeAlphanumericSegment(bits, result, count, fc1InEffect);
+                decodeAlphanumericSegment(bits, result, count, fc1InEffect, byteSegments);
```

Comment:
OK more specifically - there are no 'bytes' encoded in anything but byte mode. There are streams of bits, but I don't see much use case in returning them. Their representation is entirely specific to the QR code, not any user payload. Any use of these modes to encode arbitrary bytes would be wrong. So if you're putting raw compressed bytes into the QR code without using byte mode, that's the problem.